src/AdventureGoldGovernance.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import "../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import "../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/ERC20Permit.sol";
+import "../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/ERC20Votes.sol";
+import "../lib/openzeppelin-contracts/contracts/interfaces/IERC6372.sol";
+import "../lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import "../lib/openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @title Adventure Gold Governance
+/// @author Will Papper
+/// @notice This contract allows users to deposit AGLD tokens and receive
+/// Adventure Gold Governance tokens in return.
+/// Users can also burn Adventure Gold Governance tokens to withdraw AGLD tokens.
+/// @dev The contract inherits from ERC20Burnable, ERC20Permit, and ERC20Votes.
+contract AdventureGoldGovernance is IERC6372, ERC20Burnable, ERC20Permit, ERC20Votes {
+    using SafeERC20 for IERC20;
+
+    /// @notice Error emitted when transfers are not to or from the contract.
+    error OnlyTransfersToFromContract();
+
+    /// @notice Emitted when a user deposits AGLD tokens, minting Adventure Gold
+    /// Governance tokens.
+    /// @param user The address of the user who deposited tokens.
+    /// @param amount The amount of AGLD tokens deposited.
+    event Deposit(address indexed user, uint256 amount);
+
+    /// @notice Emitted when a user withdraws AGLD tokens, burning Adventure
+    /// Gold Governance tokens.
+    /// @param user The address of the user who withdrew tokens.
+    /// @param amount The amount of AGLD tokens withdrawn.
+    event Withdrawal(address indexed user, uint256 amount);
+
+    /// @notice The address of the AGLD token contract.
+    address public immutable AGLD_TOKEN_ADDRESS = 0x32353A6C91143bfd6C7d363B546e62a9A2489A20;
+    /// @notice The AGLD token contract.
+    IERC20 public immutable AGLD_TOKEN = IERC20(AGLD_TOKEN_ADDRESS);
+
+    /// @notice Constructor for the Adventure Gold Governance contract.
+    constructor() ERC20("Adventure Gold Governance", "AGLDGOV") ERC20Permit("Adventure Gold Governance") {}
+
+    /// @notice Deposits AGLD tokens and mints Adventure Gold Governance tokens in return.
+    /// @param amount The amount of AGLD tokens to deposit.
+    /// @dev The AGLD tokens are transferred from the msg.sender to this contract.
```

Comment:
Thoughts on stating in the `@dev` section that an approval is required for that amount before it can be deposited?

Reply:
Great call, will add!